### PR TITLE
appveyor - Add support for the latest versions of npcap via an OEM ac…

### PR DESCRIPTION
**Pull request by chmorgan**
…count, complements of Fyodor and the nmap group

### CODE REVIEW

**Comment by chmorgan** [comment]
Npcap doesn't have remote pcap support yet. Discussing it with them now. https://github.com/nmap/npcap/issues/74

File: .appveyor.yml [hunk 1]
```diff
@@ -16,19 +23,18 @@ for:
       - image: Visual Studio 2017
   install:
     - choco install -y dotnetcore-sdk --version=3.1.301
-    - choco install -y winpcap
+    - ps: .\scripts\InstallNpcap.ps1
```
Comment (kayoub5) [inline .appveyor.yml:26]: @chmorgan The only reason I used the Visual Studio 2017 image was simply to test SharpPcap with WinPcap
I think this should remain as is as a backward compatibility test

File: azure-pipelines.yml [hunk 2]
```diff
@@ -36,10 +36,28 @@ jobs:
   - script: dotnet restore -s https://api.nuget.org/v3/index.json
   - script: bash scripts/test.sh --filter
 
+- job: windows_2016_vs2017_npcap
```
Comment (kayoub5) [inline azure-pipelines.yml:39]: Visual Studio version have little to no importance in test cases, what matters really is the Windows version and driver (Npcap or WinPcap)

Comment (chmorgan) [reply azure-pipelines.yml:39]: @kayoub5 agree but also want to make sure we are testing what people are normally using. It's easy enough to add the vs2017 and older windows builds. Not sure there is a big downside.

File: .appveyor.yml (hunk 1 above)
Comment (chmorgan) [reply .appveyor.yml:26]: agree, I kept the winpcap tests

**Comment by chmorgan** [comment]
@kayoub5 mind reviewing this now that I think its in reasonable shape?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -34,12 +34,36 @@ jobs:
...
+    vmImage: 'VS2017-Win2016'
+  steps:
+  - pwsh: ./scripts/InstallNpcap.ps1
+    env:
+      npcap_oem_key: $(npcap_oem_key)
+  - script: dotnet restore -s https://api.nuget.org/v3/index.json
+  # Disable WinDivert to avoid an internal CLR error due to secure boot having been enabled
+  - script: bash scripts/test.sh --filter "TestCategory!=RemotePcap&TestCategory!=WinDivert"
 
 - job: windows_2019_vs2019
```
Comment (kayoub5) [inline azure-pipelines.yml:51]: I think we should indicate in the name that this one uses winpcap
```suggestion
- job: windows_2019_vs2019_winpcap
```

**Comment by kayoub5** [comment]
Since appveyor is very slow, I believe it should not run Performance tests in `Visual Studio 2019` job
image

**Comment by chmorgan** [comment]
@kayoub5 does it matter much if the builds take a long time or maybe I'm missing something? Typically I'll iterate pushing new changes and looking at the non-appveyor results and then once the code is stable wait for the appveyor ones to complete. For us it might be a minor pain but for contributors it would seem like that kind of delay should be acceptable, its not likely you or I might get to the code that quickly.

File: azure-pipelines.yml (hunk 3 above)
Comment (chmorgan) [reply azure-pipelines.yml:51]: I like it, let me fix.

**Comment by kayoub5** [comment]
LGTM

**Pull request merged**